assets/javascripts/modules/checkout/checkout.js

```diff
@@ -135,13 +138,25 @@ define(['$',
     };
 
     var finishAccount = function () {
-        if ($FINISH_ACCOUNT_SUBMIT.length) {
+        if ($FINISH_ACCOUNT_FORM.length) {
             bean.on($FINISH_ACCOUNT_SUBMIT[0], 'click', function (e) {
                 e.preventDefault();
 
                 if (validations.validateFinishAccount()) {
-                    bean.off($FINISH_ACCOUNT_SUBMIT[0], 'click');
-                    bean.fire($FINISH_ACCOUNT_SUBMIT[0], 'click');
+                    var action = $FINISH_ACCOUNT_FORM.attr('action');
+                    var formData = ajax.reqwest.serialize($FINISH_ACCOUNT_FORM[0]);
+
+                    ajax({
+                        url: action,
+                        method: 'POST',
+                        data: formData
+                    }).then(function () {
+                        $FINISH_ACCOUNT_FORM.hide();
+                        $ALL_DONE.show();
+
+                    }).catch(function (jsonError) {
+                        console && console.error("Error while submitting guest user password form:", jsonError);
```

Comment:
Can we do something other than console log here? Display something to the user maybe? Probably good to capture the exception with Raven too.